### PR TITLE
Add basic JSX localization framework

### PR DESCRIPTION
**Pull request by goldpbear**
Addresses: #874, #688.

Add a localization system for content rendered via JSX. We use the [i18next](https://www.i18next.com/getting-started.html) library to perform translations. Currently this system is implemented just for React content in the `williams` flavor (but see TODOs below).

This is a dead simple pass at an initial JSX localization system. To perform localization, we send a language code from the build process to the React entry point, then use it to set the locale in the `i18next` library.

All translatable content is in `src/flavors/williams/static/components/messages.js`. A similar structure would be used for other flavors (and for base messages), meaning that we'll have two files with translatable content for a given flavor (similar to the current gettext setup).

Note that no `makemessages` functionality (i.e. the ability to extract all strings for a given language in preparation for distribution to translators) is provided at this point.

**Advantages:**
* Minimal changes to build process
* Quick to set up

**Disadvantages:**
* Overall localization system will consist of content rendered both by `i18next` and `gettext`, until the React port is complete
* All language strings sent to client, regardless of selected language

### TODO
* ~~Integrate changes to remaining VV flavors (`manzanares`, `hull`, and `augusta`)~~
    * ~~note that only VV flavors have React code in `master` at this point~~

### CODE REVIEW

**Comment by goldpbear** [comment]
@zmbc -- ah, sorry. I got a little carried away with the lingo. "VV" (Village Vanguard) are the flavors we worked on during the Spain trip (`williams`, `manzanares`, `hull`, and `augusta`). They are the only flavors with React stuff currently in production.

**Comment by zmbc** [comment]
@goldpbear Gotcha.

File: src/flavors/williams/static/components/input-explorer.js [hunk 1]
```diff
@@ -105,6 +108,7 @@ class InputExplorer extends Component {
           communityInput={this.props.communityInput}
           subcategoryNames={this.subcategoryNames}
           visibility={(selectedCategory === constants.INPUT_CATEGORY_SUMMARY_NAME) ? true : false}
+          languageCode={this.props.languageCode}
```
Comment (modulitos) [inline src/flavors/williams/static/components/input-explorer.js]: I don't see this prop being used in InputExplorerSummary. Why are we passing it into that component?

File: scripts/config-loader.js [hunk 2]
```diff
@@ -72,6 +72,7 @@ module.exports = function(source) {
   const template = Handlebars.compile(templateSource);
   outputFile = template({
     config: config,
+    languageCode: "en_US"
```
Comment (modulitos) [inline scripts/config-loader.js]: Maybe I'm missing something, but wouldn't a user need to change this value to view the app in their chosen language?

For example, how would the user select to view the app in Spanish, and how would that update the languageCode here?

Comment (goldpbear) [reply scripts/config-loader.js]: So eventually I think we should move toward @zmbc's localize on the client idea, where we serve a single bundle with all localized content in it, and switching languages is just a matter of changing the `i18n` code. But as long as we still have content handled by gettext, we need to fetch a whole new localized bundle when the user switches languages.

So it's true that content in the above `messages.js` file that doesn't match the current locale is redundant. We could filter that out in the build process I guess, but I'm going on the assumption that eventually we'll want all language content available on the client.

**Comment by goldpbear** [comment]
@modulitos -- re the timing of this PR: I suppose we could merge this for all the React components currently in `master` (which right now is just the "comment explorer" feature of the VV flavors), then rebase and port the localization for the other React PRs we have open.

**Pull request merged**